File: src/main/java/io/github/welingtonmonteiro/multiplerun/ui/MultirunStatusBarWidgetFactory.java
package io.github.welingtonmonteiro.multiplerun.ui;

import org.jetbrains.annotations.NotNull;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.util.Disposer;
import com.intellij.openapi.wm.StatusBar;
import com.intellij.openapi.wm.StatusBarWidget;
import com.intellij.openapi.wm.StatusBarWidgetFactory;

/** Registers the {@link MultirunStatusBarWidget} in the IDE status bar (toggle in the widget menu). */
public class MultirunStatusBarWidgetFactory implements StatusBarWidgetFactory {

    @NotNull
    @Override
    public String getId() {
        return MultirunStatusBarWidget.ID;
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "Multiple Run";
    }

    @Override
    public boolean isAvailable(@NotNull Project project) {
        return true;
    }

    @NotNull
    @Override
    public StatusBarWidget createWidget(@NotNull Project project) {
        return new MultirunStatusBarWidget(project);
    }

    @Override
    public void disposeWidget(@NotNull StatusBarWidget widget) {
        Disposer.dispose(widget);
    }

    @Override
    public boolean canBeEnabledOn(@NotNull StatusBar statusBar) {
        return true;
    }
}

File: src/main/kotlin/io/github/welingtonmonteiro/multiplerun/ui/MultirunStatusBarWidgetFactory.kt
package io.github.welingtonmonteiro.multiplerun.ui

import com.intellij.openapi.project.Project
import com.intellij.openapi.util.Disposer
import com.intellij.openapi.wm.StatusBar
import com.intellij.openapi.wm.StatusBarWidget
import com.intellij.openapi.wm.StatusBarWidgetFactory

/** Registers the [MultirunStatusBarWidget] in the IDE status bar (toggle in the widget menu). */
class MultirunStatusBarWidgetFactory : StatusBarWidgetFactory {

    override fun getId(): String = MultirunStatusBarWidget.ID

    override fun getDisplayName(): String = "Multiple Run"

    override fun isAvailable(project: Project): Boolean = true

    override fun createWidget(project: Project): StatusBarWidget = MultirunStatusBarWidget(project)

    override fun disposeWidget(widget: StatusBarWidget) {
        Disposer.dispose(widget)
    }

    override fun canBeEnabledOn(statusBar: StatusBar): Boolean = true
}
